app/src/main/java/org/mifos/mobile/ui/fragments/ReviewLoanApplicationFragment.kt

```diff
@@ -115,38 +101,43 @@ class ReviewLoanApplicationFragment : BaseFragment() {
 
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
-        binding.tvLoanProduct.text = viewModel.getLoanProduct()
-        binding.tvLoanPurpose.text = viewModel.getLoanPurpose()
-        binding.tvPrincipalAmount.text = viewModel.getPrincipal().toString()
-        binding.tvExpectedDisbursementDate.text = viewModel.getDisbursementDate()
-        binding.tvSubmissionDate.text = viewModel.getSubmissionDate()
-        binding.tvCurrency.text = viewModel.getCurrency()
-        binding.tvNewLoanApplication.text = viewModel.getLoanName()
-        binding.tvAccountNumber.text = viewModel.getAccountNo()
-
-        binding.btnLoanSubmit.setOnClickListener {
-            showProgress()
-            viewModel.submitLoan()
-                ?.observeOn(AndroidSchedulers.mainThread())
-                ?.subscribeOn(Schedulers.io())
-                ?.subscribeWith(object : DisposableObserver<ResponseBody>() {
-                    override fun onComplete() {
-                    }
-
-                    override fun onNext(t: ResponseBody) {
-                        hideProgress()
-                        if (viewModel.getLoanState() == LoanState.CREATE) {
-                            showLoanAccountCreatedSuccessfully()
-                        } else {
-                            showLoanAccountUpdatedSuccessfully()
+        binding.apply {
+            tvLoanProduct.text = viewModel.getLoanProduct()
+            tvLoanPurpose.text = viewModel.getLoanPurpose()
+            tvPrincipalAmount.text = String.format(
+                Locale.getDefault(),
+                "%.2f", viewModel.getPrincipal(),
+            )
+            tvExpectedDisbursementDate.text = viewModel.getDisbursementDate()
+            tvSubmissionDate.text = viewModel.getSubmissionDate()
+            tvCurrency.text = viewModel.getCurrency()
+            tvNewLoanApplication.text = viewModel.getLoanName()
+            tvAccountNumber.text = viewModel.getAccountNo()
+
+            btnLoanSubmit.setOnClickListener {
+                showProgress()
+                viewModel.submitLoan()
+                    ?.observeOn(AndroidSchedulers.mainThread())
+                    ?.subscribeOn(Schedulers.io())
+                    ?.subscribeWith(object : DisposableObserver<ResponseBody>() {
+                        override fun onComplete() {
```

Comment:
this piece of code should not have been here. Move this to viewmodel and while doing so make sure to use coroutines instead of RxJava

Reply:
I changed it to Livedata now. Please review it.